src/Compilers/CSharp/Portable/Syntax/SyntaxFormatter.cs

```diff
@@ -127,8 +129,27 @@ public override SyntaxToken VisitToken(SyntaxToken token)
             }
         }
 
+        private SyntaxToken GetNextRelevantToken(SyntaxToken token)
+        {
+            // get next token, skipping zero width tokens except for end-of-directive tokens
+            var nextToken = token.GetNextToken(
+                t => SyntaxToken.NonZeroWidth(t) || t.Kind() == SyntaxKind.EndOfDirectiveToken, 
+                t => t.Kind() == SyntaxKind.SkippedTokensTrivia);
+
+            if (_consideredSpan.Contains(nextToken.FullSpan))
```

Comment:
Tag @heejaechang.  I think we might want Intersects here.  I believe try to format the "edges" of the span so that the whitespace between the previous token and the first token of the span get formatted, and between the last token and the beginning of the token after it.